examples/lines_bars_and_markers/markevery_demo.py

```diff
@@ -3,99 +3,94 @@
+cases = [
+    None,
+    8,
+    (30, 8),
+    [16, 24, 32],
+    [0, -1],
+    slice(100, 200, 3),
+    0.1,
+    0.4,
+    (0.2, 0.4)
+]
+
+# data points
 delta = 0.11
 x = np.linspace(0, 10 - 2 * delta, 200) + delta
 y = np.sin(x) + 1.0 + delta
 
-
-def trim_axs(axs, N):
-    """
-    Reduce *axs* to *N* Axes. All further Axes are removed from the figure.
-    """
-    axs = axs.flat
-    for ax in axs[N:]:
-        ax.remove()
-    return axs[:N]
-
 ###############################################################################
-# Plot each markevery case for linear x and y scales
+# markevery with linear sacles
+# ----------------------------
 
-axs = plt.figure(figsize=figsize, constrained_layout=True).subplots(rows, cols)
-axs = trim_axs(axs, len(cases))
-for ax, case in zip(axs, cases):
-    ax.set_title('markevery=%s' % str(case))
-    ax.plot(x, y, 'o', ls='-', ms=4, markevery=case)
+fig, axs = plt.subplots(3, 3, figsize=(10, 6), constrained_layout=True)
+for ax, markevery in zip(axs.flat, cases):
+    ax.set_title(f'markevery={markevery}')
+    ax.plot(x, y, 'o', ls='-', ms=4, markevery=markevery)
 
 ###############################################################################
-# Plot each markevery case for log x and y scales
-
-axs = plt.figure(figsize=figsize, constrained_layout=True).subplots(rows, cols)
-axs = trim_axs(axs, len(cases))
-for ax, case in zip(axs, cases):
-    ax.set_title('markevery=%s' % str(case))
+# markevery with log scales
+# -------------------------
+#
+# Note that the log scale causes a visual asymmetry in the marker distance for
+# the integer-based subsampling. In contrast, the float-based subsampling
+# creates even distributions, because it's based on fractions of the Axes
+# diagonal, not on data coordinates or data indices.
+
+fig, axs = plt.subplots(3, 3, figsize=(10, 6), constrained_layout=True)
+for ax, markevery in zip(axs.flat, cases):
+    ax.set_title(f'markevery={markevery}')
     ax.set_xscale('log')
     ax.set_yscale('log')
-    ax.plot(x, y, 'o', ls='-', ms=4, markevery=case)
+    ax.plot(x, y, 'o', ls='-', ms=4, markevery=markevery)
 
 ###############################################################################
-# Plot each markevery case for linear x and y scales but zoomed in
-# note the behaviour when zoomed in.  When a start marker offset is specified
-# it is always interpreted with respect to the first data point which might be
-# different to the first visible data point.
-
-axs = plt.figure(figsize=figsize, constrained_layout=True).subplots(rows, cols)
-axs = trim_axs(axs, len(cases))
-for ax, case in zip(axs, cases):
-    ax.set_title('markevery=%s' % str(case))
-    ax.plot(x, y, 'o', ls='-', ms=4, markevery=case)
+# markevery on zoomed plots
+# -------------------------
+#
+# When a start marker offset is specified it is always interpreted with
+# respect to the first data point which might be different to the first
+# visible data point.
```

Comment:
Maybe? 
```suggestion
# When a starting marker offset is specified, it counts with
# respect to the first data point, which might not be the first
# visible data point.
```

Reply:
I've rewritten the whole paragraph. This was not really getting to the point to make about zooming.